content/en/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins.md

```diff
@@ -103,9 +106,28 @@ The general workflow of a device plugin includes the following steps:
         // Plugin can run device specific operations and instruct Kubelet
         // of the steps to make the Device available in the container
         rpc Allocate(AllocateRequest) returns (AllocateResponse) {}
+
+        // GetPreferredAllocation returns a preferred set of devices to allocate
+        // from a list of available ones. The resulting preferred allocation is not
+        // guaranteed to be the allocation ultimately performed by the
+        // devicemanager. It is only designed to help the devicemanager make a more
+        // informed allocation decision when possible.
+        rpc GetPreferredAllocation(PreferredAllocationRequest) returns (PreferredAllocationResponse) {}
+
+        // PreStartContainer is called, if indicated by Device Plugin during registeration phase,
+        // before each container start. Device plugin can run device specific operations
+        // such as resetting the device before making devices available to the container.
+        rpc PreStartContainer(PreStartContainerRequest) returns (PreStartContainerResponse) {}
   }
   ```
 
+  **Note:** Plugins are not required to provide useful implementations for
+  `GetPreferredAllocation()` or `PreStartContainer()`. Flags indicating which
+  (if any) of these calls are available should be set in the `DevicePluginOptions`
+  message sent back by a call to `GetDevicePluginOptions()`. The `kubelet` will
+  always call `GetDevicePluginOptions()` to see which optional functions are
+  available, before calling any of them directly.
```

Comment:
```suggestion
{{% note %}}
Plugins are not required to provide useful implementations for
`GetPreferredAllocation()` or `PreStartContainer()`. Flags indicating which
(if any) of these calls are available should be set in the `DevicePluginOptions`
message sent back by a call to `GetDevicePluginOptions()`. The kubelet
always calls `GetDevicePluginOptions()` to see which optional functions are
available, before calling any of them directly.
{{% /note %}}
```

Use a [Note](https://kubernetes.io/docs/contribute/style/style-guide/#note) shortcode (untested).

Reply:
1. Updated per your suggestion with 1 minor difference. I used `{{< note >}}` not `{{% note %}}` as the link you provided suggested.
2. No, this actually moved to beta in 1.18, but the documentation was never updated to reflect that. The feature gate was toggled to "always-on" back before 1.18 was released.

Reply:
On detail 1, I did also file https://github.com/kubernetes/website/issues/22488 about the advice being not quite right for callout shortcodes. Anyway, it renders well enough as is.

Reply:
It looks good to me: https://deploy-preview-21607--kubernetes-io-vnext-staging.netlify.app/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins/